### PR TITLE
systemd-run: add unit and invocation_id JSON output

### PR DESCRIPTION
**Pull request by lnykryn**

### CODE REVIEW

File: src/run/run.c [hunk 1]
```diff
@@ -133,6 +135,7 @@ static int help(void) {
                "                                  STDERR\n"
                "  -P --pipe                       Pass STDIN/STDOUT/STDERR directly to service\n"
                "  -q --quiet                      Suppress information messages during runtime\n"
+               "  --json=pretty|short|off         Wrap the name of unit and invocation_id in JSON\n"
```
Comment (bluca) [inline src/run/run.c]: align the --json indentation with --quiet above so that long options are all aligned

**Review by bluca — APPROVED** [review]
Please add some validation to test/units/TEST-74-AUX-UTILS.run.sh even if superficial

**Comment by bluca** [comment]
Actually can you add support for the new option in shell-completion/bash/systemd-run as well please?

**Comment by lnykryn** [comment]
@bluca updated, PTAL.

File: man/systemd-run.xml [hunk 2]
```diff
@@ -393,6 +393,8 @@
         <xi:include href="version-info.xml" xpointer="v219"/></listitem>
       </varlistentry>
 
+      <xi:include href="standard-options.xml" xpointer="json" />
```
Comment (YHNdnzj) [inline man/systemd-run.xml]: Please group standard options down below

File: src/run/run.c [hunk 3]
```diff
@@ -1686,6 +1697,33 @@ static int chown_to_capsule(const char *path, const char *capsule) {
         return chmod_and_chown(path, 0600, st.st_uid, st.st_gid);
 }
 
+static int print_unit_invocation(char *unit, sd_id128_t invocation_id) {
```
Comment (YHNdnzj) [inline src/run/run.c]: const char *

File: src/run/run.c [hunk 4]
```diff
@@ -1686,6 +1697,33 @@ static int chown_to_capsule(const char *path, const char *capsule) {
...
+        _cleanup_(sd_json_variant_unrefp) sd_json_variant *v = NULL;
+        int r;
+
+        assert(unit);
+
+        if (FLAGS_SET(arg_json_format_flags, SD_JSON_FORMAT_OFF)) {
+                if(sd_id128_is_null(invocation_id))
```
Comment (YHNdnzj) [inline src/run/run.c]:
```suggestion
                if (sd_id128_is_null(invocation_id))
```

**Review by YHNdnzj — APPROVED** [review]

File: src/run/run.c [hunk 5]
```diff
@@ -1807,15 +1845,15 @@ static int start_transient_service(sd_bus *bus) {
         }
 
         if (!arg_quiet) {
-                sd_id128_t invocation_id;
+                sd_id128_t invocation_id = SD_ID128_NULL;
```
Comment (yuwata) [inline src/run/run.c:1848]: Why initialized here? I do not think it is necessary.

File: src/run/run.c [hunk 6]
```diff
@@ -1976,7 +2014,7 @@ static int start_transient_scope(sd_bus *bus) {
         _cleanup_strv_free_ char **env = NULL, **user_env = NULL;
         _cleanup_free_ char *scope = NULL;
         const char *object = NULL;
-        sd_id128_t invocation_id;
+        sd_id128_t invocation_id = SD_ID128_NULL;
```
Comment (yuwata) [inline src/run/run.c:2017]: Here also.

File: src/run/run.c [hunk 7]
```diff
@@ -1686,6 +1697,33 @@ static int chown_to_capsule(const char *path, const char *capsule) {
...
+                        log_info("Running as unit: %s; invocation ID: " SD_ID128_FORMAT_STR, unit, SD_ID128_FORMAT_VAL(invocation_id));
+                return 0;
+        }
+
+        r = sd_json_variant_set_field_string(&v, "unit", unit);
+        if (r < 0)
+                return r;
+
+        if (!sd_id128_is_null(invocation_id)) {
+                r = sd_json_variant_set_field_string(&v, "invocation_id", SD_ID128_TO_STRING(invocation_id));
```
Comment (yuwata) [inline src/run/run.c:1719]: Maybe better to introduce `sd_json_variant_set_field_id128()` and `_set_field_uuid()`.

**Comment by yuwata** [comment]
-> #33990.

**Pull request merged**